src/main/java/org/extism/sdk/chicory/ChicoryModule.java

```diff
@@ -36,6 +37,11 @@ static Instance.Builder instanceWithOptions(Instance.Builder m, Manifest.Options
         if (opts.aot && !IS_NATIVE_IMAGE_AOT) {
             m.withMachineFactory(aotMachineFactory);
         }
+        if (opts.memoryLimits != null) {
+            m.withMemoryFactory(limits -> {
+                return new ByteArrayMemory(limits);
+            }).withMemoryLimits(opts.memoryLimits);
+        }
```

Comment:
I am not 100% sure but I think you only need to set one of these?

Reply:
memory = requireNonNullElse(memoryFactory, ByteBufferMemory::new)
                                    .apply(requireNonNullElse(memoryLimits, defaultLimits));

Reply:
so it is a bit different, since I force the usage of ByteArrayMemory instead of ByteBufferMemory